src/mds/MDCache.cc

```diff
@@ -13597,16 +13600,16 @@ void MDCache::dispatch_quiesce_inode(const MDRequestRef& mdr)
     return;
   }
 
-  if (in->get_projected_inode()->get_quiesce_block()) {
-    dout(10) << __func__ << " quiesce is blocked for this inode; dropping locks!" << dendl;
-    mdr->mark_event("quiesce blocked");
-    mds->locker->drop_locks(mdr.get());
-    /* keep authpins! */
-    qs.inc_inodes_blocked();
-    mdr->internal_op_finish->complete(0);
-    mdr->internal_op_finish = nullptr;
-    return;
-  }
+
+  // since we're holding onto the quiesce lock,
+  // we don't need the other capabilities related locks anymore
+  // as the quiesce mask will prevent any non-shared capabilities from being issued
+  // see CInode::get_caps_quiesce_mask()
+
+  mds->locker->drop_lock(mdr.get(), &in->filelock);
+  mds->locker->drop_lock(mdr.get(), &in->authlock);
+  mds->locker->drop_lock(mdr.get(), &in->linklock);
+  mds->locker->drop_lock(mdr.get(), &in->xattrlock);
```

Comment:
Should we also drop the version lock? Or rather, I think we should, are there any reasons we shouldn't?